### DISCUSSION TITLE
add a link parameter in table

### DISCUSSION THREAD

**Discussion by flosigime** [sqlpage/SQLPage#33] [Ideas]
As we can use a list with a link parameter, the same could be really useful in a table, especially for dynamic component from database : you can see your data in the table and with a link,go to a form and modify it if you need...

**Comment by lovasoa** [comment]
Hi !

You can already add a link inside a table my using the `markdown` parameter, see the last example in https://sql.ophir.dev/documentation.sql?component=table#component

However, this adds the link to a single table cell. We could add a new `_sqlpage_link` property to table rows, but that would be a little tricky to handle, since HTML does not support nesting a table row in a link anchor element.

**Reply by flosigime** [reply]
found it. I didn't succeded in using a dynamic parameter in markdown. Just need to concatenate...

```
select 'table' as component,
'modify' as markdown;
select 
'[link](name_of_page.sql?='||field_name|| ')' as modify,
field1 as name_of_column, 
field2 as name_of_column from name_of_the_table;
```

**Reply by lovasoa** [reply]
Yes, that's it. Ideally, field_name would be a numeric identifier (that cannot contain special characters such as `)` or `&`).

And name_of_column is what gets displayed to the user, so ideally you would use `field1 as "name of column"` instead (with quotes).
